src/Ticket.php

```diff
@@ -4409,6 +4409,14 @@ public function showForm($ID, array $options = [])
             $options['_noupdate'] = true;
         }
 
+        // compute priority if is new ticket and user have no right to change priority
+        if ($this->isNewItem() && !$canpriority) {
+            $this->fields['priority'] = $this->computePriority(
+                $this->fields['urgency'] ?: $predefined_fields['urgency'] ?? 3 /* Medium */,
+                $this->fields['impact'] ?: $predefined_fields['impact']  ?? 3 /* Medium */
+            );
+        }
```

Comment:
This seems a bit redundant with the code present in `setPredefinedFields()`: https://github.com/glpi-project/glpi/blob/329b50bd6a9f1b5c96c035bd294d23484f7a4cc2/src/CommonITILObject.php#L803-L815

The same kind of code will be executed twice when rendering the ticket form. I guess there is something that could be simplified here (but, again, this code is a mess, so I am not sure of anything).